examples/nlp/multi_label_classification.py

```diff
@@ -0,0 +1,382 @@
+print(f"Original label: {sample_label}")
+
+label_binarized = mlb.transform([sample_label])
+print(f"Label-binarized representation: {label_binarized}")
+
+"""
+## Data preprocessing and `tf.data.Dataset` objects
+
+We first get percentile estimates of the sequence lengths. The purpose will be clear in a
+moment.
+"""
+
+train_df["summaries"].apply(lambda x: len(x.split(" "))).describe()
+
+"""
+Notice that 50% of the abstracts have a length of 154 (you may get a different number
+based on the split). So, any number near that is a good enough approximate for the
+maximum sequence length.
+
+Now, we write utilities to prepare our datasets that would go straight to the text
+classifier model.
+"""
+
+max_seqlen = 150
+batch_size = 128
+padding_token = "<pad>"
+
+
+def unify_text_length(text, label):
+    # Split the given abstract and calculate its length.
+    word_splits = tf.strings.split(text, sep=" ")
+    sequence_length = tf.shape(word_splits)[0]
+
+    # Calculate the padding amount.
+    padding_amount = max_seqlen - sequence_length
+
+    # Check if we need to pad or truncate.
+    if padding_amount > 0:
+        unified_text = tf.pad([text], [[0, padding_amount]], constant_values="<pad>")
+        unified_text = tf.strings.reduce_join(unified_text, separator="")
+    else:
+        unified_text = tf.strings.reduce_join(word_splits[:max_seqlen], separator=" ")
+
+    # The expansion is needed for subsequent vectorization.
+    return tf.expand_dims(unified_text, -1), label
+
+
+def make_dataset(dataframe, is_train=True):
+    label_binarized = mlb.transform(dataframe["terms"].values)
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (dataframe["summaries"].values, label_binarized)
+    )
+    dataset = dataset.shuffle(batch_size * 10) if is_train else dataset
+    dataset = dataset.map(unify_text_length).cache()
+    return dataset.batch(batch_size).prefetch(tf.data.AUTOTUNE)
+
+
+"""
+Now we can prepare the `tf.data.Dataset` objects. 
+"""
+
+train_dataset = make_dataset(train_df, is_train=True)
+validation_dataset = make_dataset(val_df, is_train=False)
+test_dataset = make_dataset(test_df, is_train=False)
+
+"""
+## Dataset preview
+"""
+
+text_batch, label_batch = next(iter(train_dataset))
+
+for i, text in enumerate(text_batch[:5]):
+    label = label_batch[i].numpy()[None, ...]
+    print(f"Abstract: {text[0]}")
+    print(f"Label(s): {mlb.inverse_transform(label)[0]}")
+    print(" ")
+
+"""
+## Vocabulary size for vectorization
+
+Before we feed the data to our model we need to represent them as numbers. For that
+purpose, we will use the
+[`TextVectorization` layer](https://keras.io/api/layers/preprocessing_layers/text/text_vectorization).
+It can operate as a part of your main model so that the model is excluded from the core
+preprocessing logic. This greatly reduces the chances of training and serving skew.
+
+We first calculate the number of unique words present in the abstracts.
```

Comment:
If you want to index all words, you don't need to precompute that number. You can just pass `max_tokens=None`. However it's usually a good idea to limit `max_tokens` to ~20k for better efficiency.